src/sagemaker/amazon/amazon_estimator.py

```diff
@@ -93,8 +93,14 @@ def __init__(
             enable_network_isolation=enable_network_isolation,
             **kwargs
         )
-        data_location = data_location or "s3://{}/sagemaker-record-sets/".format(
-            self.sagemaker_session.default_bucket()
+
+        data_location = data_location or (
+            s3_utils.s3_path_join_with_end_slash(
+                "s3://",
+                self.sagemaker_session.default_bucket(),
+                self.sagemaker_session.default_bucket_prefix,
```

Comment:
Is this backward compatible where earlier the prefix was sagemaker-record-sets and now there is one more layer in between.?

Reply:
yep, behavior will stay exactly the same as now unless and until the Session's default_bucket_prefix is set with a non None/empty value (can see s3_path_join unit tests for cases Ive tested)